src/common/data_compression.cpp

```
@@ -0,0 +1,56 @@
+// Copyright 2019 yuzu Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <lz4hc.h>
+
+#include "data_compression.h"
+
+namespace Compression {
+
+std::vector<u8> CompressDataLZ4(const u8* source, std::size_t source_size,
+                                bool use_LZ4_high_compression) {
+    if (source_size > LZ4_MAX_INPUT_SIZE) {
```

This should probably be an assertion rather than silently returning an empty vector, since it's usually a caller screwing up in this case.